pgmpy/tests/test_estimators/test_TreeAugmentedNaiveBayesSearch.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+
+import pandas as pd
+
+from pgmpy.estimators import TreeAugmentedNaiveBayesSearch
+
+
+class TestTreeAugmentedNaiveBayesSearch(unittest.TestCase):
```

Comment:
I think it would a good idea to add some more realistic tests. For example, generate data from a model and then try to learn it back the structure.

Reply:
Since TAN is generally used for classification tasks, perhaps the assert statement should measure classification errors?

The steps are:

1. Manually build a TAN graph and CPDs
2. Generate N rows of data from graph
3. Learn the TAN graph from data
4. Use the learned graph to predict from the independent variables of the data
5. Measure classification errors

I don’t suppose you have code examples along this line already that I can bootstrap from?

(I think this might go nicely into `examples` as a demo as well.)

Reply:
Yes, something along those lines. But instead of testing for classification errors, I would suggest testing for smaller parts. For example, testing the learned structure or the learned parameters on a given structure. The problem with testing classification error directly (starting from a initial model/generated data) is that there are multiple steps where uncertainty can be introduced and it all can add up to give unreliable tests (or tests failing sometimes).